### PR TITLE
feat: Add codemirror syntax highlighting back in and move to a static color

### PR DESCRIPTION
**Pull request by geoffgscott**
For some reason the ::selection does not seem to accept our .dark-mode, .light-mode variables. For now we will hard code a color. 

This also adds syntax highlighting back in.

### CODE REVIEW

**Comment by amritk** [comment]
should we just move to prism?

**Comment by marclave** [comment]
> should we just move to prism?

im down for this! I still say let's merge this since it fixes a bunch of bugs and we can do follow up work to remove the couple of lang  highlighting we have : )

File: packages/use-codemirror/src/hooks/useCodeMirror.ts [hunk 1]
```diff
@@ -283,11 +298,25 @@ function getCodeMirrorExtensions({
...
+    extensions.push(
+      indentOnInput(),
+      bracketMatching(),
+      autocompletion(),
+      closeBrackets(),
+      keymap.of([
+        ...completionKeymap,
+        ...closeBracketsKeymap,
+        selectAllKeyBinding,
+      ]),
```
Comment (hanspagel) [inline packages/use-codemirror/src/hooks/useCodeMirror.ts:313]: What are these doing? I can’t see any indent, bracket, autocomplete … magic when I play around with the editor. 🤔

**Review by hanspagel — APPROVED** [review]
Awesome PR! I’ve added some minor details that came to mind when reviewing.

Anyway, I’m too stupid to see what the added CodeMirror plugins do. If you could enlighten me, that would be great.

But let’s get this in!

**Comment by hanspagel** [comment]
Ah, and I just saw @marclave had another fix for the selection color here: https://github.com/scalar/scalar/blob/3108786309e4cd6daaa7854a7cc1a5819fa702ed/packages/use-codemirror/src/hooks/useCodeMirror.ts#L260-L263

Should we add this to the PR here instead of hard coding the color? 🤔

File: packages/use-codemirror/src/hooks/useCodeMirror.ts (hunk 1 above)
Comment (marclave) [reply packages/use-codemirror/src/hooks/useCodeMirror.ts:313]: maybe do a fresh build?

**Comment by marclave** [comment]
> Ah, and I just saw @marclave had another fix for the selection color here:
> 
> https://github.com/scalar/scalar/blob/3108786309e4cd6daaa7854a7cc1a5819fa702ed/packages/use-codemirror/src/hooks/useCodeMirror.ts#L260-L263
> 
> Should we add this to the PR here instead of hard coding the color? 🤔

woah i had no idea this was in main! we noticed css variables dont work in selection selectors

**Comment by geoffgscott** [comment]
> Ah, and I just saw @marclave had another fix for the selection color here:
> 
> https://github.com/scalar/scalar/blob/3108786309e4cd6daaa7854a7cc1a5819fa702ed/packages/use-codemirror/src/hooks/useCodeMirror.ts#L260-L263
> 
> Should we add this to the PR here instead of hard coding the color? 🤔

This approach does not work. I think just stale code in Marcs branch

**Review by marclave — APPROVED** [review]

**Pull request merged**